### PR TITLE
Node styling improvements

### PR DESCRIPTION
**Pull request by georgefst**
This is a cleaned-up version of the bulk of #980/#987.

I've fixed up comments and commit messages, and rebased to fold last commit in earlier. I also fixed a bug where edges out of `THole` were invisible - this turned out to be due to a typo (`stroke-red-tertiaryy`).

### CODE REVIEW

File: src/components/TreeReactFlow/Flavor.ts [hunk 1]
```diff
@@ -522,3 +529,72 @@ export const boxFlavorBackground = (flavor: NodeFlavorBoxBody): string => {
       return "bg-yellow-primary";
   }
 };
+
+/** What sort of node does this flavor correspond to?
+ * Note that the backend could in principal tell us this independently of flavors,
+ * since it comes down to whether the node ultimately comes from an `Expr`, `Type` or `Kind`.
+ */
+export const isTypeLevel = (flavor: NodeFlavor): "term" | "type" | "kind" => {
```
Comment (georgefst) [inline src/components/TreeReactFlow/Flavor.ts:537]: Is there a good name for "type or term or kind"? The existing naming here was just about okay when we weren't considering kinds, but now it's misleading.

Comment (brprice) [reply src/components/TreeReactFlow/Flavor.ts:537]: I would probably use "sort". It is used this way in Pure Type Systems.

**Comment by dhess** [comment]
My Internet is currently brutally bad, so I won't be able to meaningfully review this. So long as it's more or less in line with #987, I'll be OK with it.

**Comment by georgefst** [comment]
> My Internet is currently brutally bad, so I won't be able to meaningfully review this. So long as it's more or less in line with #987, I'll be OK with it.

There's no difference in functionality, just some minor code changes. I'd merge, but CI is blocked due to some Vault issue.

**Pull request merged**